plugins/serializers/prometheusremotewrite/prometheusremotewrite.go

```diff
@@ -338,6 +359,179 @@ func getPromTS(name string, labels []prompb.Label, value float64, ts time.Time,
+		return false, nil
+	}
+	schema, ok := fields["schema"]
+	if !ok {
+		return false, nil
+	}
+	schemaInt, ok := schema.(int64)
+	if !ok {
+		return false, nil
+	}
+	counterResetHint, ok := fields["counter_reset_hint"]
+	if !ok {
+		return false, nil
+	}
+	counterResetHintInt, ok := counterResetHint.(uint64)
+	if !ok {
+		return false, nil
+	}
+	zeroThreshold, ok := fields["zero_threshold"]
+	if !ok {
+		return false, nil
+	}
+	zeroThresholdFloat, ok := zeroThreshold.(float64)
+	if !ok {
+		return false, nil
+	}
+	zeroCount, ok := fields["zero_count"]
+	if !ok {
+		return false, nil
+	}
+	zeroCountFloat, ok := zeroCount.(float64)
+	if !ok {
+		return false, nil
+	}
+
+	floatHistogram := &histogram.FloatHistogram{
+		Count:            countFloat,
+		Sum:              sumFloat,
+		Schema:           int32(schemaInt),
+		CounterResetHint: histogram.CounterResetHint(counterResetHintInt),
+		ZeroThreshold:    zeroThresholdFloat,
+		ZeroCount:        zeroCountFloat,
+		PositiveSpans:    make([]histogram.Span, 0),
+		NegativeSpans:    make([]histogram.Span, 0),
+		PositiveBuckets:  make([]float64, 0),
+		NegativeBuckets:  make([]float64, 0),
+	}
+
+	fmt.Println(fmt.Sprintf("floatHistogram alright so far: %v", floatHistogram))
+
+	// expand positiveSpans and negativeSpans into fields
+	i := 0
+	for {
+		offset, ok := fields[fmt.Sprintf("positive_span_%d_offset", i)]
+		if !ok {
+			break
+		}
+		offsetInt, ok := offset.(int64)
+		if !ok {
+			break
+		}
+		length, ok := fields[fmt.Sprintf("positive_span_%d_length", i)]
+		if !ok {
+			break
+		}
+		lengthInt, ok := length.(uint64)
+		if !ok {
+			break
+		}
+		positiveSpan := histogram.Span{
+			Offset: int32(offsetInt),
+			Length: uint32(lengthInt),
+		}
+		floatHistogram.PositiveSpans = append(floatHistogram.PositiveSpans, positiveSpan)
+		i++
+	}
+	i = 0
+	for {
+		offset, ok := fields[fmt.Sprintf("negative_span_%d_offset", i)]
+		if !ok {
+			break
+		}
+		offsetInt, ok := offset.(int64)
+		if !ok {
+			break
+		}
+		length, ok := fields[fmt.Sprintf("negative_span_%d_length", i)]
+		if !ok {
+			break
+		}
+		lengthInt, ok := length.(uint64)
+		if !ok {
+			break
+		}
+		negativeSpan := histogram.Span{
+			Offset: int32(offsetInt),
+			Length: uint32(lengthInt),
+		}
+		floatHistogram.NegativeSpans = append(floatHistogram.NegativeSpans, negativeSpan)
+		i++
+	}
+	i = 0
+	for {
+		bucket, ok := fields[fmt.Sprintf("positive_bucket_%d", i)]
+		if !ok {
+			break
+		}
+		bucketFloat, ok := bucket.(float64)
+		if !ok {
+			break
+		}
+		floatHistogram.PositiveBuckets = append(floatHistogram.PositiveBuckets, bucketFloat)
+		i++
+	}
+	i = 0
+	for {
+		bucket, ok := fields[fmt.Sprintf("negative_bucket_%d", i)]
+		if !ok {
+			break
+		}
+		bucketFloat, ok := bucket.(float64)
+		if !ok {
+			break
+		}
+		floatHistogram.NegativeBuckets = append(floatHistogram.NegativeBuckets, bucketFloat)
+		i++
+	}
+	err := floatHistogram.Validate()
+	if err != nil {
+		return false, nil
+	}
+	return true, floatHistogram
```

Comment:
Dropping the bool as suggested above you can do
```suggestion
	return floatHistogram.Validate()
```

Reply:
make sense